### ISSUE TITLE
mission "Destroy the FLF base!" won't finish.

### ISSUE THREAD

**Issue by amylaar** [naev/naev#1677]
This is Pilot Terry on his mission to combat the FLF terrorists.
I have infiltrated the FLF base to buy their maps, and then under the pretence to fight a Dvaered patrol (OK, a few ships might have experienced some hull existence discontinuity before I could speak to someone in authority), I grouped up with the Dvaered in Arcanis.
Commanding the USZH "Heap of Trouble" (1), I assisted the HDSF Obstinate in destroying the FLF base by keeping the FLF fighters at bay (or rather away from the Obstinate's vulnerable fighter bays).
After re-fueling at Raelid Outpost and changing ships in Anubis, I landed at Stalwart Station for de-briefing.  Yet somehow, even after the de-briefing, the mission seemed to linger.
Scientists that theorize that our universe is but a simulation claim to have received the below quoted test from outside the
fabric of our space-time.

(1): USZH: Underhandedly Stolen Za'lek Hephaestus

Naev v0.8.0+1186.gd31e7790d (linux-x86_64)
...
Reached main menu
Warning: [misn_runFunc] Mission 'Disrupt a Dvaered Patrol' -> 'returnFLFControl': [string "missions/flf/flf_pre02.lua"]:336: Pilot is invalid.
stack traceback:
	[C]: in function 'control'
	[string "missions/flf/flf_pre02.lua"]:336: in function <[string "missions/flf/flf_pre02.lua"]:334>
Warning: [hook_runMisn] Hook [p_death] '1097' -> 'returnFLFControl' failed
Warning: [misn_runFunc] Mission 'Disrupt a Dvaered Patrol' -> 'returnFLFControl': [string "missions/flf/flf_pre02.lua"]:336: Pilot is invalid.
stack traceback:
	[C]: in function 'control'
	[string "missions/flf/flf_pre02.lua"]:336: in function <[string "missions/flf/flf_pre02.lua"]:334>
Warning: [hook_runMisn] Hook [p_death] '1183' -> 'returnFLFControl' failed
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [widget_exists] window '933' does not exist
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'iarMissions' not found in window '933'!
Warning: [iar_getWidget] Widget 'iarMissions' not found
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'iarMissions' not found in window '933'!
Warning: [iar_getWidget] Widget 'iarMissions' not found
Warning: [window_wget] Window '933' not found in list!
Warning: [widget_exists] window '933' does not exist
Warning: [window_wget] Window '933' not found in list!
Warning: [widget_exists] window '933' does not exist
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [widget_exists] window '933' does not exist
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'btnApproach' not found in window '933'!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'txtPortrait' not found in window '933'!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'imgPortrait' not found in window '933'!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'imgPortraitBG' not found in window '933'!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'txtMission' not found in window '933'!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_wget] Window '933' not found in list!
Warning: [window_getwgt] Widget 'iarMissions' not found in window '933'!
Warning: [iar_getWidget] Widget 'iarMissions' not found
Warning: [misn_runFunc] Mission 'Destroy the FLF base!' -> 'land': [string "missions/dvaered/dv_antiflf03.lua"]:223: attempt to index global 'time' (a number value)
stack traceback:
	[string "missions/dvaered/dv_antiflf03.lua"]:223: in function <[string "missions/dvaered/dv_antiflf03.lua"]:210>
Warning: [hook_runMisn] Hook [land] '1246' -> 'land' failed

**Comment by UncombedCoconut** [comment]
I've pushed a fix (will not help a saved game where the mission is in progress).
It addresses the very last error message (final 4 lines of your log), which caused the problem.

All of the preceding error messages look super sketchy, but I'm not sure what would have caused them. Did you notice any related funny business before starting the dv_antiflf03 mission?
If not, I suppose we can close the issue.

**Comment by amylaar** [comment]
> I've pushed a fix (will not help a saved game where the mission is in progress).

Thanks.  I've adjusted the savefile with gunzip / vi / gzip to use the new variable name, and the mission finished after
taking off and landing again.

> It addresses the very last error message (final 4 lines of your log), which caused the problem.
> 
> All of the preceding error messages look super sketchy, but I'm not sure what would have caused them.

I had a look at how returnFLFControl is invoked, at it seems it's trying yo do something with the FLF ships when I destroy them.
What is supposed to happen there?

No clue about the other stuff.

**Comment by UncombedCoconut** [comment]
> > I've pushed a fix (will not help a saved game where the mission is in progress).
> 
> Thanks. I've adjusted the savefile with gunzip / vi / gzip to use the new variable name, and the mission finished after
> taking off and landing again.

:)

> > It addresses the very last error message (final 4 lines of your log), which caused the problem.
> > All of the preceding error messages look super sketchy, but I'm not sure what would have caused them.
> 
> I had a look at how returnFLFControl is invoked, at it seems it's trying to do something with the FLF ships when I destroy them.
> What is supposed to happen there?

By my reading, it's for when *they* kill *you*. (Not that that ever would have happened. ;))
The idea is to release them from scripted manual control (drop everything and attack the player) so you can cinematically blow up while they go about their business.
For all I know, that message reflects something harmless happening (like an attempt to do that for a Frontier fleet member who died).

> No clue about the other stuff.

It looks like some GUI stuff (on the landing screen) gone horribly wrong, but I don't know how/why.

**Comment by amylaar** [comment]
> By my reading, it's for when _they_ kill _you_. (Not that that ever would have happened. ;))

I had just realized that having only 26 reputation at the start of the mission was cutting it a bit fine if the objective was to get
hailed by the Dvaered before I wipe them all out, and was paying too little attention to the ship movement and the shield booster...

> The idea is to release them from scripted manual control (drop everything and attack the player) so you can cinematically blow up while they go about their business.
> For all I know, that message reflects something harmless happening (like an attempt to do that for a Frontier fleet member who died).

Yes, that makes perfect sense.

**Comment by bobbens** [comment]
Pushed a fix to the manual control warnings at the top. The GUI issue should be opened in a separate issue if it persists. I haven't been able to reproduce. Closing as the FLF stuff seems to be done. Thanks!
